### PR TITLE
Recent updates to CMake, GNU dirs, VS versions

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

**Review by derobins — APPROVED** [review]

File: CMakeInstallation.cmake [hunk 1]
```diff
@@ -364,7 +364,7 @@ if (NOT HDF4_EXTERNALLY_CONFIGURED AND NOT HDF4_NO_PACKAGES)
     set (CPACK_PACKAGING_INSTALL_PREFIX "/${CPACK_PACKAGE_INSTALL_DIRECTORY}")
     set (CPACK_PACKAGE_ICON "${HDF_RESOURCES_EXT_DIR}/hdf.icns")
 
-    option (HDF4_PACK_MACOSX_FRAMEWORK  "Package the HDF Library in a Framework" OFF)
+    option (HDF4_PACK_MACOSX_FRAMEWORK  "Package the HDF Library in a Frameworks" OFF)
```
Comment (derobins) [inline CMakeInstallation.cmake:367]: Is "Frameworks" correct here? 'a' is singular.

Comment (byrnHDF) [reply CMakeInstallation.cmake:367]: IDK, but it is used everywhere.

File: config/cmake/libh4cc.in [hunk 2]
```diff
@@ -27,6 +27,27 @@ prg=$dir/$(basename -- "$prg") || exit
 printf '%s\n' "$prg"
 printf 'dir is %s\n' "$dir"
 
+
+# Show the configuration summary of the library recorded in the
+# libhdf5.settings file residing in the lib directory.
```
Comment (lrknox) [inline config/cmake/libh4cc.in:32]: comment should be libhdf4.settings?

File: config/cmake_ext_mod/HDFLibMacros.cmake [hunk 3]
```diff
@@ -13,12 +13,14 @@ macro (EXTERNAL_JPEG_LIBRARY compress_type jpeg_pic)
...
             -DJPEG_EXTERNALLY_CONFIGURED:BOOL=OFF
             -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
+            -DCMAKE_DEBUG_POSTFIX:STRING=${CMAKE_DEBUG_POSTFIX}
             -DCMAKE_INSTALL_PREFIX:PATH=${CMAKE_INSTALL_PREFIX}
             -DCMAKE_RUNTIME_OUTPUT_DIRECTORY:PATH=${CMAKE_RUNTIME_OUTPUT_DIRECTORY}
             -DCMAKE_LIBRARY_OUTPUT_DIRECTORY:PATH=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
             -DCMAKE_ARCHIVE_OUTPUT_DIRECTORY:PATH=${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}
             -DCMAKE_PDB_OUTPUT_DIRECTORY:PATH=${CMAKE_PDB_OUTPUT_DIRECTORY}
             -DCMAKE_ANSI_CFLAGS:STRING=${jpeg_pic}
+            -DHDF_USE_GNU_DIRS:STRING=${HDF5_USE_GNU_DIRS}
```
Comment (lrknox) [inline config/cmake_ext_mod/HDFLibMacros.cmake:23]: Is HDF5_USE_GNU_DIRS correct for HDF 4?
(several instances in this file)

Comment (byrnHDF) [reply config/cmake_ext_mod/HDFLibMacros.cmake:23]: Fixing hdf5 refs in 6 different places.

**Review by lrknox — APPROVED** [review]

**Review by derobins — APPROVED** [review]

**Pull request merged**